lib/rules/require-fetch-import.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const ERROR_MESSAGE = 'Explicitly import `fetch` instead of using `window.fetch`';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'enforce explicit import for `fetch()`',
+      category: 'Miscellaneous',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/require-fetch-import.md',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    let fetchImportFound = false;
+    const fetchCalls = [];
+
+    return {
+      ImportDeclaration(node) {
+        // search for a `fetch` import
+        for (const specifier of node.specifiers) {
+          if (specifier.type === 'ImportDefaultSpecifier' || specifier.type === 'ImportSpecifier') {
+            if (specifier.local.type === 'Identifier' && specifier.local.name === 'fetch') {
+              fetchImportFound = true;
+            }
+          }
+        }
+      },
+
+      CallExpression(node) {
```

Comment:
There's a much easier and more robust way to implement this rule using the [ReferenceTracker](https://eslint-utils.mysticatea.dev/api/scope-utils.html#referencetracker-class) from eslint-utils. Using that class, you can easily flag a violation on usage of the global `fetch` function. We do the same thing in this rule:

https://github.com/ember-cli/eslint-plugin-ember/blob/master/lib/rules/prefer-ember-test-helpers.js

This approach will also ensure you don't flag `fetch()` when `function fetch() { ... }` is defined in the same file. 

Can you try this better approach? It's only a few lines of code.

Reply:
done